src/Rules/MissingTypehintCheck.php

```diff
@@ -132,4 +138,27 @@ public function getNonGenericObjectTypesWithGenericClass(Type $type): array
 		return $objectTypes;
 	}
 
+	/**
+	 * @param \PHPStan\Type\Type $type
+	 * @return \PHPStan\Type\Type[]
+	 */
+	public function getCallablesWithMissingPrototype(Type $type): array
+	{
+		if (!$this->checkMissingCallablePrototype) {
+			return [];
+		}
+
+		$result = [];
+		TypeTraverser::map($type, static function (Type $type, callable $traverse) use (&$result): Type {
+			if (
+				($type instanceof CallableType && $type->isCommonCallable()) ||
+				($type instanceof ObjectType && $type->getClassName() === \Closure::class)) {
```

Comment:
appears that this causes some undesirable behaviour with generics when `T of Closure` is used (is that even supposed to work to begin with?)

Reply:
@ondrejmirtes is there a better way to check for this already in place? I don't feel comfortable with this condition.

I feel like it would be more consistent to have `Closure` and `Callable` types both have something like `isCommonCallable()`, but the problem is that common closures are just created as `ObjectType` and I don't know where it should be changed.

This check works but it feels dirty.